test/test_coldcard.py

```diff
@@ -0,0 +1,252 @@
+assert(enum_res[0]['path'] == '/tmp/ckcc-simulator.sock')
+assert(enum_res[0]['fingerprint'] == '0f056943')
+
+# Test path + type
+logging.info('Testing path and type not specified')
+gmxp_res = process_commands(['getmasterxpub'])
+assert('error' in gmxp_res)
+assert(gmxp_res['error'] == 'You must specify a device type or fingerprint for all commands except enumerate')
+assert('code' in gmxp_res)
+assert(gmxp_res['code'] == -1)
+
+logging.info('Testing path and type specified')
+gmxp_res = process_commands(['-t', enum_res[0]['type'], '-d', enum_res[0]['path'], 'getmasterxpub'])
+assert(gmxp_res['xpub'] == 'tpubDDpWvmUrPZrhSPmUzCMBHffvC3HyMAPnWDSAQNBTnj1iZeJa7BZQEttFiP4DS4GCcXQHezdXhn86Hj6LHX5EDstXPWrMaSneRWM8yUf6NFd')
+
+# Test fingerprint autodetect
+logging.info('Testing fingerprint autodetect')
+gmxp_res = process_commands(['-f', enum_res[0]['fingerprint'], 'getmasterxpub'])
+assert(gmxp_res['xpub'] == 'tpubDDpWvmUrPZrhSPmUzCMBHffvC3HyMAPnWDSAQNBTnj1iZeJa7BZQEttFiP4DS4GCcXQHezdXhn86Hj6LHX5EDstXPWrMaSneRWM8yUf6NFd')
+
+# Test device type autodetect
+logging.info('Testing only device type specified')
+gmxp_res = process_commands(['-t', enum_res[0]['type'], 'getmasterxpub'])
+assert(gmxp_res['xpub'] == 'tpubDDpWvmUrPZrhSPmUzCMBHffvC3HyMAPnWDSAQNBTnj1iZeJa7BZQEttFiP4DS4GCcXQHezdXhn86Hj6LHX5EDstXPWrMaSneRWM8yUf6NFd')
+
+# Test getxpub
+# BIP 32 test vectors
+logging.info('Testing getxpub and getmasterxpub')
+with open(os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data/coldcard_xpubs.json'), encoding='utf-8') as f:
+    vectors = json.load(f)
+for vec in vectors:
+    # Test getmasterxpub
+    gmxp_res = process_commands(dev_args + ['getmasterxpub'])
+    assert(gmxp_res['xpub'] == vec['master_xpub'])
+
+    # Test the path derivs
+    for path_vec in vec['vectors']:
+        gxp_res = process_commands(dev_args + ['getxpub', path_vec['path']])
+        assert(gxp_res['xpub'] == path_vec['xpub'])
+
+# signtx and getkeypool need bitcoind, so start that
+logging.info('Setting up bitcoind')
+datadir = tempfile.mkdtemp()
+bitcoind_proc = subprocess.Popen([args.bitcoind, '-regtest', '-datadir=' + datadir, '-noprinttoconsole'])
+def cleanup_bitcoind():
+    bitcoind_proc.kill()
+    shutil.rmtree(datadir)
+atexit.register(cleanup_bitcoind)
+# Wait for cookie file to be created
+while not os.path.exists(datadir + '/regtest/.cookie'):
+    time.sleep(0.5)
+# Read .cookie file to get user and pass
+with open(datadir + '/regtest/.cookie') as f:
+    userpass = f.readline().lstrip().rstrip()
+rpc = AuthServiceProxy('http://{}@127.0.0.1:18443'.format(userpass))
+
+# Wait for bitcoind to be ready
+ready = False
+while not ready:
+    try:
+        rpc.getblockchaininfo()
+        ready = True
+    except JSONRPCException as e:
+        time.sleep(0.5)
+        pass
+
+# Setup bitcoind with no privkey wallet and some blocks
+rpc.generatetoaddress(101, rpc.getnewaddress())
+rpc.createwallet('trezor_test', True)
```

Comment:
s/trezor_test/coldcard_test/ everywhere in this file

Reply:
Fixed